Teams - main branch (Make PRs against this branch!) (#140)
mulesoft/docs-access-management: modules/ROOT/pages/business-groups.adoc

Business groups are self-contained resource groups that contain Anypoint Platform resources such as applications and APIs. Business groups provide a way to separate and control access to Anypoint Platform resources because users have access only to the business groups in which they have a role.

> Business groups reside within the root organization and are organized in a hierarchical tree where the top-level business group is the root. Each business group you create has one direct parent and can have multiple children.

Reviewer: Can I just point out that the top of a tree is not the root? lol

Author: Lol. Fair.

[NOTE]
Business groups are not enabled by default in a new Anypoint Platform account. To activate business groups in your organization, contact your MuleSoft representative. There is a limit of 100 business groups per organization.

> If you are have the Organization Administrator permission and your organization is enabled for business group support, you can create business groups within your organization. A user with the Organization Administrator permission is also the administrator of any business groups within the organization.

Reviewer: ```suggestion
If you have the Organization Administrator permission and your organization is enabled for business group support, you can create and manage business groups within your organization.
```

Reviewer: By "your organization" do we mean the root or some child of the root?

Business groups provide more fine-grained control over access to resources.

Because business groups are hierarchical, the owner of a parent business group automatically has and retains administrator permissions for any child business group of that parent, even it they make another Organization Administrator user owner of a child business group.

Conversely, owners of child business groups cannot:

* Access or modify the parent business group or root organization settings
* View the parent business group's client ID and client secret

When you create a business group under another business group, only the redistributable entitlements, such as VPCs and load balancers, that were assigned to the parent business group can be allocated to the child business group.

== Create a Business Group

> When you create a new business group under the top-level business group, all the current users with the Organization Administrator permission in the top-level business group appear in the list of users in the newly created business group.

Reviewer: Are we using "top-level business group" in the same way as "parent business group"?

Reviewer: Is it important to specify "current" for the users that appear in the list?

Author: ```suggestion
When you create a new business group under the top-level, or _root_, business group, all the current users with the Organization Administrator permission in the root business group appear in the list of users in the newly created business group.
```

Author: Yes, as users added later are not automatically added, to my knowledge.

include::partial$include-nav-steps-org-admin.adoc[]

include::partial$include-nav-business-groups.adoc[]

> . To create the first business group for your organization, click *Create business group*.

Reviewer: first or top-level or parent?
If "first," this step still applies to any time your create a business group, yes? So do we need "first"?

Author: ```suggestion
. To create the root business group for your organization, click *Create business group*.
```

> . To create a child business group, next to the business group under which you want to create a new business group, click the *...* menu.

Reviewer: Or...?
```suggestion
. To create a child business group, click the *...* menu next to the business group that will be the parent.
```

. Click *Create child group*.
. In the dialog box, enter:
  * *Business Group name*: Name for the new business group.

>   * *Owner*: Assign an existing user, with the Organization Administrator permission, as the business group owner.

Reviewer: I know that this wording is very precise, but will this work just as well?
```suggestion
  * *Owner*: Assign an existing Organization Administrator as the business group owner.
```

Create a business group hierarchy to help you better control user access to resources.

> . Click the *...* menu next to of the business group in which you want to create the child.

Reviewer: Compare with Line 35 and standardize.

Author: ```suggestion
 . To create a child business group, click the *...* menu next to the business group that will be the parent.
```

When you log in to Anypoint Platform, you return to the business group where you were last active when you logged out of Anypoint Platform.
--

> If you do not have the Organization Administrator permission, you can view only the business groups that you have permissions to view. In the *Organization* tab, the tree of your organization displays only the business groups you are a member of.

Reviewer: ```suggestion
If you do not have the Organization Administrator permission, you can view only the business groups that you have permissions to view. In the *Organization* tab, your organization tree displays only the business groups to which you belong.
```

[[creatingroles]]
=== Roles and Membership in Business Groups

> Roles can be applied at the root organization level as well as at the business group level. However, each controls different resources. Roles and permissions assigned to a user in one particular business group are independent of the roles and permissions assigned to that same user in other business groups.

Reviewer: To remove the modal verb....
```suggestion
Roles are applied at the root organization level as well as at the business group level. However, each controls different resources. Roles and permissions assigned to a user in one particular business group are independent of the roles and permissions assigned to that same user in other business groups.
```

> To belong to a business group, you must be granted a role within that business group. Once you are visible in a business group, you can be granted individual permissions in that business group. All users, regardless of which business groups they are in, belong to the root organization and can be assigned individual permissions in the root organization.

Reviewer 2: hmm... this section is strange. 

```suggestion
To interact with a business group, you must be granted a permission within that business group, or be a member of a team that is granted permission within that business group. All users, regardless of which business groups they are in, belong to the root organization.
```

or we could say team or role, but that feels wordy

> Members that are added to a business group can view and access the business group, but to access certain resources, you must have the appropriate roles assigned to you. For example, to access APIs and CloudHub deployments in the business group, you must be assigned roles that belong to that business group.

Reviewer 2: `Members that are added to a business group can view` -> `Members that are granted permissions in a business group can view`

Author: ```suggestion
Members granted permissions in a business group can view and access the business group, but to access certain resources, you must have the appropriate permissions or roles assigned to you. For example, to access APIs and CloudHub deployments in the business group, you must be assigned roles that belong to that business group.
```

> Members granted permissions in a business group can view and access the business group, but to access certain resources, you must have the appropriate permissions or roles assigned to you. For example, to access APIs and CloudHub deployments in the business group, you must be assigned roles that belong to that business group.

Reviewer: Are the "certain resources" not part of the business group? The second half of the first sentence does not seem to follow on the first.

Reviewer: Aside from that, we might want to make the first part of that sentence address "you" in the same way as the second part:
```suggestion
As a member of a business group, you are automatically granted permissions to view and access the business group, but to access certain resources, you must have the appropriate permissions or roles assigned to you. For example, to access APIs and CloudHub deployments in the business group, you must be assigned roles that belong to that business group.
```

Author: No, your assessment is correct.

> For example, if you have the Organization Administrator permission in BusinessGroupA, you can grant users access to APIs and CloudHub deployments in BusinessGroupA.

Reviewer 2: Could we specify users and teams?

Author: ```suggestion
For example, if you have the Organization Administrator permission in BusinessGroupA, you can grant users and teams access to APIs and CloudHub deployments in BusinessGroupA.
```

If you also have the Organization Administrator permission in BusinessGroupB, this does not mean that you can grant access to APIs and CloudHub deployments in BusinessGroupA to users in BusinessGroupB and vice versa.

> Resources that belong to the root organization require roles specified at the root organization level.

Reviewer 2: ```suggestion
Resources that belong to the root organization require permissions specified at the root organization level.
```

> If you also have the Organization Administrator permission in BusinessGroupB, this does not mean that you can grant access to APIs and CloudHub deployments in BusinessGroupA to users in BusinessGroupB and vice versa.

Reviewer: ```suggestion
If you also have the Organization Administrator permission in BusinessGroupB, this does not mean that you can grant access to APIs and CloudHub deployments in BusinessGroupA to users in BusinessGroupB, and vice versa.
```

> Users who are assigned the Organization Administrator permission in the root organization, or any business group, are automatically assigned that same role in any child business group created within that business group, so are automatically visible in the child business group. This does not apply to users who are assigned the Organization Administrator permission after a child business group has been created.

Reviewer 2: I'd like to revisit this section with a few examples, but don't think it is urgent.

> Users who are assigned the Organization Administrator permission in the root organization, or any business group, are automatically assigned that same role in any child business group created within that business group, so are automatically visible in the child business group. This does not apply to users who are assigned the Organization Administrator permission after a child business group has been created.

Reviewer: ```suggestion
Users who are assigned the Organization Administrator permission at any level are automatically assigned that same role in any child business group created within that level, so are automatically visible in the child. This does not apply retroactively to users who are assigned the Organization Administrator permission after a child is created.
```

=== Delete a Business Group

Only an organization administrator that belongs to a business group can delete it. The top-level (root) business group can't be deleted, even by an organization administrator.

include::partial$include-nav-organizations.adoc[]

. Click the *...* menu next to the business group to delete.

> . In the confirmation dialog, enter the name of the business group, then click *Delete*.

Reviewer: ```suggestion
. In the confirmation dialog, enter the name of the business group and then click *Delete*.
```

[[redistribute-resources]]
== Redistribute Resources Between Existing Business Groups

Sometimes you need to redistribute resources (such as vCores, VPCs, load balancers, and so on) based on deployment of apps in your business groups.
The business group hierarchy prevents a child business group from consuming resources from either its parent group or the root organization unless an Organization Administrator redistributes those resources. Resource distribution proceeds from root to parent to child.

> For example, a child business group cannot receive sandbox vCores if the parent business group does not have any sandbox vCores available, even if the root organization has sandbox vCores available.

Reviewer: Do we need "sandbox"? It is later init. capped, as an environment name.

Additionally, the root organization or parent business group cannot reclaim resources that are currently allocated to a business group that is lower in the hierarchy.

You must have the Organization Administrator permission to redistribute resources.

include::partial$include-nav-business-groups.adoc[]

. Click the *Settings* tab.
. Enter the number of vCores you want the business group to have.
. Click *Save changes*.

> The *Business Group Info* window includes a resource counter showing a minimum value that equals the amount of the resource that is reassigned to child business groups plus the amount of the resource currently in use by the business group. Unavailable resources are not represented by a slider in the window.

Reviewer 2: ```suggestion
The *Business Group Info* window includes a resource counter showing a minimum value that equals the amount of the resource that is reassigned to child business groups plus the amount of the resource currently in use by the business group. Unavailable resources are not represented in the window.
```

> The *Business Group Info* window includes a resource counter showing a minimum value that equals the amount of the resource that is reassigned to child business groups plus the amount of the resource currently in use by the business group. Unavailable resources are not represented in the window.

Reviewer: How is this a "minimum" value?

Reviewer: I wonder if this would be more clear if we inserted the meaning of the second sentence into the first:
```suggestion
The *Business Group Info* window includes a resource counter that shows a value representing only available resources, the sum of those resources reassigned to child business groups plus those currently in use by the parent business group. 
```

Reviewer: ...although it seems to me that resources in use by the parent would be considered unavailable....

Author: ```suggestion
The *Business Group Info* window includes a resource counter that shows a value representing only available resources, the sum of those resources reassigned to child business groups plus those currently in use by the selected business group. 
```
